code/datums/components/vision_hurting.dm

```diff
@@ -0,0 +1,27 @@
+/// A component that damages eyes that look at the owner
+/datum/component/vision_hurting
+	var/damage_per_second
+	var/message
+
+/datum/component/vision_hurting/Initialize(damage_per_second=1, message="Your eyes burn as you look at")
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.damage_per_second = damage_per_second
+	src.message = message
+
+	START_PROCESSING(SSdcs, src)
```

Comment:
would it be worthwhile to add this to the SSmob processing list instead of SSdcs or is that taboo? feels like it would be more integrated to the whole damage function we have here and all of that in case admins are trolling.

Reply:
We can add it to whichever, I feel like it's better for SSmob to handle primarily mobs and throwing other stuff in there would taint stats but if we want to put this in something that isn't SSdcs I see nothing wrong with that. As for synchronizing process calls, ideally we shouldn't be depending on that in our code so if something here causes problems in that regard I'd see it as a good thing and the base issue should be fixed instead if possible.